### ISSUE TITLE
Investigate how maps work in instrumentation-ui

### ISSUE THREAD

**Issue by brian428** [USACE/water-ui-old#89]
Look into how the [instrumentation-ui](https://github.com/USACE/instrumentation-ui) project uses maps. In particular, I would look at the maps-bundle, instrument-map-bundle, instrument-group-map-bundle, and explore-map-bundle. You can clone the project but unfortunately you probably can't run it as I believe it depends on APIs that we don't have access to.

I'm sure we can learn a lot from looking at these. I'm fairly sure we'll need to do some of the same things, such as potentially having more than one map bundle (e.g. one for the main map page, a small map to show on a location detail page just for that location, and possibly a map showing the USACE districts on the home page). I also think we'll need some way to store the "state" of the main map, so that if the user expands the location detail to fill the page, then contracts it again, that the map could be restored to the previous extent and zoom level. 

It also looks like we could use some of the map view state logic from the @corpsmap/corpsmap-bundles/createOlMapBundle (under our node_modules). For example, at "moveend" it stores the new center coordinates, zoom level, etc.

Basically, spend some time looking at the existing USACE map-related code and see if we can learn from and use any of it.

**Comment by mohitb3** [comment]
Seems like `maps-bundle` loads the default map. This bundle contains selectors/action creators which are shared by all the different maps, such as `doMapsInitialize` and `doMapsShutdown`. We have our own `maps-bundle` file but it's not being used.

The other maps bundles, `instrument-map-bundle`, `instrument-group-map-bundle`, `explore-map-bundle`, add layers for each specific map. For example the `instrument-map-bundle` adds and instrument layer to the map, while the `explore-map-bundle` adds an explore layer to the map.

I don't think we can copy/paste their code but we can use it as a basis for ours. We already have the `locations-map-bundle` which loads the location details overlay on the map. We can create a `region-map-bundle` for loading the region layers, or a `homepage-map-bundle` if we need a separate map for the homepage. If we want to store the zoom level and other map coordinates, we can do it in the `maps-bundle` so that it's not specific to a particular map, that way if the user switches maps their map position will still be the same.
